gnomad_qc/v3/resources/basics.py

```diff
@@ -30,46 +94,51 @@ def get_gnomad_v3_mt(
     :param test: Whether to use the test MT instead of the full v3 MT
     :return: gnomAD v3 dataset with chosen annotations and filters
     """
+    logger.warning(
+        "The raw v3.1 sparse MatrixTable has been deleted. This function now calls "
+        "'get_gnomad_v3_vds' and turns the VariantDataset it into a sparse MatrixTable."
+        " We recommend direct use of the VariantDataset by calling 'get_gnomad_v3_vds' "
+        "instead."
+    )
+
     if test:
         mt = gnomad_v3_testset.mt()
-    else:
-        mt = gnomad_v3_genotypes.mt()
-
-    if key_by_locus_and_alleles:
-        mt = hl.MatrixTable(
-            hl.ir.MatrixKeyRowsBy(
-                mt._mir, ["locus", "alleles"], is_sorted=True
-            )  # Prevents hail from running sort on genotype MT which is already sorted by a unique locus
-        )
-
-    if remove_hard_filtered_samples:
-        mt = mt.filter_cols(
-            hl.is_missing(
-                hard_filtered_samples.versions[CURRENT_VERSION].ht()[mt.col_key]
+        if key_by_locus_and_alleles:
+            # Prevents hail from running sort on genotype MT which is already sorted
+            # by a unique locus
+            mt = hl.MatrixTable(
+                hl.ir.MatrixKeyRowsBy(mt._mir, ["locus", "alleles"], is_sorted=True)
             )
+        if remove_hard_filtered_samples:
+            mt = mt.filter_cols(hl.is_missing(hard_filtered_samples.ht()[mt.col_key]))
+        if samples_meta or release_only:
+            meta_ht = meta.ht()
+            if samples_meta:
+                mt = mt.annotate_cols(meta=meta_ht[mt.col_key])
+            if release_only:
+                mt = mt.filter_cols(meta_ht[mt.col_key].release)
+        if split:
+            mt = mt.annotate_rows(
+                n_unsplit_alleles=hl.len(mt.alleles),
+                mixed_site=(hl.len(mt.alleles) > 2)
+                & hl.any(lambda a: hl.is_indel(mt.alleles[0], a), mt.alleles[1:])
+                & hl.any(lambda a: hl.is_snp(mt.alleles[0], a), mt.alleles[1:]),
+            )
+            mt = hl.experimental.sparse_split_multi(mt, filter_changed_loci=True)
+    else:
+        vds = get_gnomad_v3_vds(
+            split=split,
+            release_only=release_only,
+            samples_meta=samples_meta,
```

Comment:
should this also include "remove_hard_filtered_samples= remove_hard_filtered_samples"?